### DISCUSSION TITLE
Simplified Chinese rather than traditional?

### DISCUSSION THREAD

**Discussion by lucasjinreal** [openai/whisper#277] [Ideas]
```
whisper ../../data/test_data/16k16bit.wav --language Chinese --model tiny
100%|█████████████████████████████████████| 72.1M/72.1M [00:09<00:00, 8.15MiB/s]
[00:00.000 --> 00:04.800] 黃維至立於把數字世界帶入每個人每個家庭每個組織
[00:04.800 --> 00:19.800] 購劍萬物互聯帶智能世界

```

this is not very accurate.

**Comment by jongwook** [comment]
The tiny model doesn't perform well on many non-English languages. I'd recommend using the medium or large model for better performance in Chinese.

Regarding the simplified/traditional distinction, we used a single language code `zh` for all varieties of Chinese, although there can be different dialects/languages/styles of the transcripts. I'd expect the training data had both simplified and traditional scripts, and the model should be capable of transcribing in both.

For example, by prompting in a certain style (i.e. `--initial_prompt` with a sentence that may have come before what's spoken in the audio), you can steer the model to transcribe [the same audio file](https://datasets-server.huggingface.co/assets/google/fleurs/--/cmn_hans_cn/train/0/audio/audio.wav) in either traditional or simplified Chinese, as well as to write with punctuations:

```
$ whisper --language Chinese --model large audio.wav
[00:00.000 --> 00:08.000] 如果他们使用航空的方式运输货物在某些航线上可能要花几天的时间才能卸货和通关
$ whisper --language Chinese --model large audio.wav --initial_prompt "以下是普通話的句子。"  # traditional
[00:00.000 --> 00:08.000] 如果他們使用航空的方式運輸貨物,在某些航線上可能要花幾天的時間才能卸貨和通關。
$ whisper --language Chinese --model large audio.wav --initial_prompt "以下是普通话的句子。"  # simplified
[00:00.000 --> 00:08.000] 如果他们使用航空的方式运输货物,在某些航线上可能要花几天的时间才能卸货和通关。
```

**Reply by ErzhongHoo** [reply]
works for me！

**Reply by wqw547243068** [reply]
Does that mean the following code with Python?

```py
prompt='以下是普通话的句子'
result = model.transcribe(audioFile, task='translate',language='zh',verbose=True,initial_prompt=prompt)
```

**Reply by vopbs** [reply]
> 微型模型在许多非英语语言上表现不佳。我建议使用中型或大型型号以获得更好的中文性能。
> 
> 关于简体/繁体的区分，我们对所有中文变体使用单一的语言代码`zh`，尽管可以有不同的方言/语言/抄本风格。我希望训练数据既有简化脚本也有繁体脚本，并且模型应该能够在这两种脚本中进行转录。
> 
> 例如，通过以某种风格提示（即`--initial_prompt`在音频中所说的内容之前可能出现的句子），您可以引导模型以繁体或简体中文转录[相同的音频文件](https://datasets-server.huggingface.co/assets/google/fleurs/--/cmn_hans_cn/train/0/audio/audio.wav)，以及用标点符号：
> 
> ```
> $ whisper --language Chinese --model large audio.wav
> [00:00.000 --> 00:08.000] 如果他们使用航空的方式运输货物在某些航线上可能要花几天的时间才能卸货和通关
> $ whisper --language Chinese --model large audio.wav --initial_prompt "以下是普通話的句子。"  # traditional
> [00:00.000 --> 00:08.000] 如果他們使用航空的方式運輸貨物,在某些航線上可能要花幾天的時間才能卸貨和通關。
> $ whisper --language Chinese --model large audio.wav --initial_prompt "以下是普通话的句子。"  # simplified
> [00:00.000 --> 00:08.000] 如果他们使用航空的方式运输货物,在某些航线上可能要花几天的时间才能卸货和通关。
> ```
I used your whole method, and I found a problem that he knew the initial beginning, which was different from the original traditional Chinese meaning.

**Reply by WorkAlbertX** [reply]
It works!Thank you very much!

**Reply by pto2k** [reply]
Thank you! 
That works and I also noticed that, the punctuation in the initial_prompt changed formating of the transcription.

I wonder if there are other things initial_prompt could affect?

--initial_prompt "以下是普通话的句子"
[00:00.000 --> 00:05.000] 我有一首歌词 题目字母 歌词要求原创
[00:05.000 --> 00:10.000] 这个歌词根据你的能力可长可短
[00:10.000 --> 00:13.000] 参与对象是所有的同学
[00:13.000 --> 00:16.000] 那么我申请了一个班级的邮箱
[00:16.000 --> 00:20.000] 这个邮箱的七个字母分别是
[00:20.000 --> 00:22.000] 风华初手字母
[00:22.000 --> 00:26.000] 然后是起航中队手字母
[00:26.000 --> 00:28.000] 便于大家去记
[00:28.000 --> 00:33.000] 风华初起航中队这七个字的手字母大家可以记一下
[00:33.000 --> 00:37.000] 那你的电子稿一定要写好名字和学号
[00:37.000 --> 00:41.000] 便于我们收集的同学进行统计
[00:41.000 --> 00:46.000] 歌词上交的时间是1月31日之前

--initial_prompt "以下是普通话的句子。"
[00:00.000 --> 00:13.000] 一首歌词,题目字母,歌词要求原创,这个歌词根据你的能力可长可短,参与对象是所有的同学。
[00:13.000 --> 00:28.000] 那么我申请了一个班级的邮箱,这个邮箱的七个字母分别是风华初手字母,然后是启航中队手字母,便于大家去记,
[00:28.000 --> 00:43.000] 那你的电子狗一定要写好名字和学号,便于我们收集的同学进行统计。歌词上交的时间是1月31日之前。

**Comment by Francoyy** [comment]
I think it is very strange that the initial prompt is the solution to fix the Simplified VS Traditional Chinese.
It does not make too much sense to me for a 2 reasons:
- Only the character 話/话 is different. The rest of the sentence is identical. Why do we have to type the whole sentence then? If the goal is to place a context, then only 1 character should be enough? like `--initial_prompt "話"`. Are we free to write whatever we want in this prompt? Also I want to note that there are many characters that are identical in both Simplified and Traditional. So some sentences would look exactly the same. Example `今天很冷` is the same in Traditional and Simplified. So it forces the user to choose carefully characters that are not identical between Simplified and Traditional Chinese.
- In Taiwan where Traditional Chinese is used, nobody says `普通話` to talk about Mandarin. People say `中文` or `國語`. Therefore it's very strange to me that you have to say `--initial_prompt "以下是普通話的句子。"` to mean "I want traditional Chinese", in a sentence that nobody says when in a traditional chinese context. 

I think that the correct way to solve Traditional/Simplified should be to use --language and choose one of the many [zh-hk, zh-cn, zh-sg, zh-tw] just like defined in the ISO 639-1 standard language codes.

It's not just a small variation.
Simplified VS Traditional is not a 1:1 mapping, it's a 1:n mapping.

Many traditional characters are all written the same in Simplified. Therefore if Simplified is outputted, it's not possible to programmatically change to traditional character. It depends on the context of the sentence.

**Reply by phineas-pta** [reply]
the primordial problem is that training data isn't properly labeled, so in order to use ISO language code, u have to fine tune, or worse re-train the model from scratch

i think the handy solution is to post process the transcript to convert between traditional and simplified

**Reply by Francoyy** [reply]
I don't mind post-processing, but like I mentioned, it is not a 1:1 mapping. If the output is Simplified, there is **no way** to recover Traditional characters with a basic mapping (lots of websites offer such a mapping, which is totally fine if you map Traditional to Simplified, but makes a lot of mistakes if you try to map Simplified to Traditional).

So in some way, it would make more sense that the output is always Traditional Chinese by default. And Whisper could include a Traditional>Simplified mapping, which would be very welcome by more than 1 billion Simplified Chinese speakers.

**Reply by Francoyy** [reply]
More here: https://en.wikipedia.org/wiki/Ambiguities_in_Chinese_character_simplification

**Reply by Francoyy** [reply]
Example, if you see the simplified character 斗, in equivalent Traditional Chinese it could be either 斗 or it could be 鬥. 
How do we know which one? A mapping algorithm wouldn't know. It needs to understand completely the context of the sentence in order to do that. And Whisper happens to be the tool that actually understands contexts of sentences.

**Reply by phineas-pta** [reply]
i understand your concerns but whisper was fed with badly formatted train data from the beginning, to fix it the way you want requires as i said finetune or re-train

AI is only as good as the data we put into them

implementing any language processing with whisper will just break it, we'll never manage to cover all edge cases for any language

**Reply by jongwook** [reply]
On the first question -- the model is not instruction fine-tuned like ChatGPT was, so we want to "engineer" the prompt to resemble the sentence that would've come before the input audio to steer the text outputs. Yet we don't want to constrain it too much on certain topic or vocabulary, or in a strange mode where there was a single letter `"話"` in the line preceding the audio. That was the motivation for using the generic sentence above. I don't speak Chinese (although I recognize some of the letters) and I used Google Translate to come up with the prompt, so that's why the phrase sounds unnatural. It's just one example prompt that worked for the example audio files that I had, but please feel free to try different prompts, which may work better!

**Reply by no1xsyzy** [reply]
Simplified vs Traditional is not even 1:n, it is sometimes n:m, e.g.
"to offend" is 干/干 , voicing gan53
"dry" is 干/乾 , voicing gan55
"sky" (occult usage) is 乾/乾 qian35

**Comment by BackMountainDevil** [comment]
In language option, there are zh and Chinese. As [vopbs](https://github.com/vopbs) said "关于简体/繁体的区分，我们对所有中文变体使用单一的语言代码zh".

So what is the diff between zh and Chinese?

And it is quite smart. I use `whisper --model large-v2 -output_dir audio/srt --output_format srt --verbose False --language zh audio/audio.m4a` to extract word. And it knows my spoken is zh-cn while it knows Huang Zhizhong uses zh-tw. In my example, I use zh as language, so should I use zh or Chinese when I want to output zh-cn?  or just add `--initial_prompt "以下是普通话的句子。"`

**Reply by LeeHaha314** [reply]
From the tokenizer you can see ‘zh’ and ‘chinese’ are actually the same, so you should add --initial_prompt to refer to zh-cn.

**Reply by Francoyy** [reply]
@BackMountainDevil I was not talking about the difference between "zh" and "Chinese", but the fact that there is "Simplified Chinese" and "Traditional Chinese". 
Their full notation is usually "zh-TW" (or zh-HK) for traditional chinese, whereas simplified chinese is "zh-Cn" or sometimes more commonly "zh".
The spoken part is the same, but the way it is written is using different characters. You could say that generally, traditional chinese contains more strokes, for example Turtle is 龟 in simplified Chinese, and 龜 in traditional.
But as @no1xsyzy mentioned, it's not a 1:1 mapping, but n:m mapping. So once Whisper outputs Chinese text, there's no way to use a script to automatically translate from simplified to traditional, or vice versa.
Hence the question if it is possible in some way to tell whisper that we would like Simplified or Traditional as output.

**Reply by LeeHaha314** [reply]
@Francoyy You got a good point, but actually the function of initial prompt is not just to switch between simplified/tradtional mode. Since both simplified and traditional texts are included in the whisper training dataset, the model should be capable for learning the n:m mapping in traditional Chinese (if there are enough patterns in the dataset).

**Reply by Francoyy** [reply]
Maybe i’m not explaining well enough, but you have to trust me on this. 
The same exact sentence spoken by someone can be written in Simplified and Traditional and there is no right or wrong. Both would be absolutely correct. 
My videos are for the Taiwan market therefore I want to have subtitle displayed in Traditional Chinese. If my videos were for China, I would want the same subtitles written in Simplified. Maybe it is more clear that way?
So it’s not about whether Whisper can detect if the subtitles should be in Simplified or Traditional, it is a personal output choice. And i want to have it in Traditional 100% of the time. hence this request.

**Reply by LeeHaha314** [reply]
@Francoyy I get you. Considering their strategy of data processing and dataset source (mainly unsupervised), I think it's hard to identify two mode clearly with the model itself , thus the usage of initial prompt could be a middle ground.

**Reply by BackMountainDevil** [reply]
guys. `--initial_prompt "以下是普通话的句子"`  tested not work on large-v3 sometimes. I want zh-cn(Simplified Chinese) but got Traditional Chinese. Do you have the same wired thing?

**Reply by chleeab** [reply]
Is it possible to separate "simplified Chinese" and "traditional Chinese" into two set of language?

**Comment by BackMountainDevil** [comment]
Got it. I think add the prompt is a good solution.&nbsp;


------------------ Original ------------------
From: Francoyy ***@***.***&gt;
Date: 周六,8月 5,2023 12:52
To: openai/whisper ***@***.***&gt;
Cc: Kearney ***@***.***&gt;, Mention ***@***.***&gt;
Subject: Re: [openai/whisper] Simplified Chinese rather than traditional? (Discussion #277)





 
@BackMountainDevil I was not talking about the difference between "zh" and "Chinese", but the fact that there is "Simplified Chinese" and "Traditional Chinese".
 Their full notation is usually "zh-TW" (or zh-HK) for traditional chinese, whereas simplified chinese is "zh-Cn" or sometimes more commonly "zh".
 The spoken part is the same, but the way it is written is using different characters. You could say that generally, traditional chinese contains more strokes, for example Turtle is 龟 in simplified Chinese, and 龜 in traditional.
 But as @no1xsyzy mentioned, it's not a 1:1 mapping, but n:m mapping. So once Whisper outputs Chinese text, there's no way to use a script to automatically translate from simplified to traditional, or vice versa.
 Hence the question if it is possible in some way to tell whisper that we would like Simplified or Traditional as output.

**Comment by cuichungman** [comment]
It is advised to label Cantonese as "yue" and Mandarin as "cmn," in accordance with their respective language codes. Continually using "zh" may result in messing up with different languages of Chinese.

**Comment by Francoyy** [comment]
Yeah, too bad… because it’s just a hack but not a guarantee. Most people
don’t understand what Simplified Chinese and Traditional Chinese is… i
guess the ultimate solution for us would be a script using google translate
and converting Traditional and Simplified, after Whisper is done generating
the subtitle…

Kearney ***@***.***>於 2023年12月11日 週一，11:03寫道：

**Comment by jbang2004** [comment]
mark

**Reply by BackMountainDevil** [reply]
try subscribe

**Comment by Francoyy** [comment]
Unfortunately, no. In terms of audio, it's the same. People have trouble to
understand that.
How could I explain it easily. It's a little bit like deciding to write in
small case, or to decide to write in UPPER CASE.
In China, people use simplified chinese characters to write characters. But
for example in Hong Kong, Taiwan they use traditional chinese characters.
So for the same video, if my audience was in Taiwan, I would want to output
traditional characters. But if my audience is from China, then I want to
get simplified characters.

**Reply by jchenny7** [reply]
How about if we specify in language whether to output zh-CN (Simplified Chinese) or zh-TW (Traditional Chinese) instead of consolidate as zh?

**Comment by Francoyy** [comment]
Yes, that makes sense to me. That was the original request. But according
to the devs, that's not how Whisper work. Therefore there is a workaround
of providing a sample input to influence the output...

**Comment by changtimwu** [comment]
I encountered a similar issue where Whisper kept outputting simplified Chinese despite setting the `initial_prompt` to "以下是普通話的句子".  I found Whipser tend to work in that way when the voices are with certain accents.   I finally found a solution:
```
initial_prompt = "以下是普通話的句子,請以繁體輸出"
```
I suspect that the Whisper model is just too effective at detecting Chinese accents. ;)

**Reply by Peter5Chiou** [reply]
thank you!

**Comment by dengchengxifrank** [comment]
mark

**Reply by no1xsyzy** [reply]
try subscribe

**Comment by mrmuke** [comment]
Just postprocess and use a package to convert traditional to simplified lol

**Comment by peterburk** [comment]
Just adding a comment here because I'd like to request Whisper to transcribe Taiwanese (Hokkien, Min Nan). 
Specifically, videos like this one:
https://archive.org/details/awakening-pray-taiwan-590963938282336

Ideally I'm aiming for the subtitles to be passed into [pingtype.github.io](pingtype.github.io) for real-time romanisation for use in singing along in church. 

The deadline is when Facebook deletes their Live Videos, on 2025-05-22.

**Reply by Francoyy** [reply]
Peter, the transcription you want to achieve is already doable with a similar command:
whisper video.mp4 --language Chinese --model large-v2 --initial_prompt "台灣的教會聯合，喚醒禱告會"

As mentioned by other people, the initial_prompt command is able to help whisper generate the correct output, in this case Traditional Chinese characters. Per my testing, it doesn't matter if the person talks in Mandarin, or Taiwanese (Hokkien, Min Nan), whisper is able to generate the correct subtitles (which is impressive). I made a video a while ago with people speaking only Taiwanese, and the command above worked beautifully.

**Reply by peterburk** [reply]
<img width="1920" alt="Screenshot 2025-03-08 at 8 14 00 pm" src="https://github.com/user-attachments/assets/70b02255-5a7d-4f00-b50c-8281e5d2b9e0" />

Thank you Francoyy! I'm trying that prompt now. 

The transcription doesn't seem to match the hard-subs though. 

Is there an AI tool to pull the yellow subtitles out of the video feed instead? I feel like that might be more accurate. 

Or maybe use both the subtitles and whisper audio recognition to get the correct sounds and timings. Just thinking out loud here.

**Reply by Francoyy** [reply]
Whisper is a tool that does audio transcription, meaning that it transforms audio into text. Whisper does not analyze the video images. It will just generate whatever text it believes has been pronounced. 
If you want a tool to analyze and extract the hard-coded subtitles from the video and generate a separate SRT file, Whisper is not the right tool, and I don't know if such a tool exists.
[Another thing that sadly if there's music on top of the voice, the subtitle generation quality will suffer.]

**Reply by peterburk** [reply]
Audio transcription might be helped by passing the video through spleeter:
[https://github.com/deezer/spleeter](https://github.com/deezer/spleeter)
I can try that :)

Video hard-subs are difficult. I've been trying 
[https://github.com/victorjoh/hard-subs-to-srt](https://github.com/victorjoh/hard-subs-to-srt)
and it flattens the video to monochrome, then passes it to Tesseract. But there's quite a lot of errors, especially compared to Google Translate.

**Reply by peterburk** [reply]
Tried spleeter as pre-processing. Getting different output, but still not the same as the subtitles. 
Maybe I should do it manually, it seems like AI hasn't reached AGI yet. 
<img width="1920" alt="Screenshot 2025-03-08 at 10 38 53 pm" src="https://github.com/user-attachments/assets/9c83f2f5-e0c8-43fa-a85d-b35407ba225c" />

**Comment by ml-inory** [comment]
I use zhconv package to convert traditional Chinese to simplified Chinese, like this:
`sim_zh = zhconv.convert(text, 'zh-hans')`
